code/modules/psionics/faculties/coercion.dm

```diff
@@ -0,0 +1,318 @@
+	cost =           15
+	cooldown =       100
+	use_melee =      TRUE
+	use_ranged =     TRUE
+	min_rank =       PSI_RANK_MASTER
+	use_description = "Target the arms or hands on disarm intent to use a ranged attack that may rip the weapons away from the target."
+
+/datum/psionic_power/coercion/spasm/invoke(var/mob/living/user, var/mob/living/carbon/human/target)
+	if(!istype(target))
+		return FALSE
+
+	if(!(user.zone_selected in list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_PRECISE_L_HAND, BODY_ZONE_PRECISE_R_HAND)))
+		return FALSE
+
+	. = ..()
+
+	if(.)
+		to_chat(user, "<span class='danger'>You lash out, stabbing into \the [target] with a lance of psi-power.</span>")
+		to_chat(target, "<span class='danger'>The muscles in your arms cramp horrendously!</span>")
+		if(prob(75))
+			target.emote("scream")
+		if(prob(75) && target.held_items[1] && target.dropItemToGround(target.get_item_for_held_index(1)))
+			target.visible_message("<span class='danger'>\The [target] drops what they were holding as their left hand spasms!</span>")
+		if(prob(75) && target.held_items[2] && target.dropItemToGround(target.get_item_for_held_index(2)))
+			target.visible_message("<span class='danger'>\The [target] drops what they were holding as their right hand spasms!</span>")
+		return TRUE
+
+/datum/psionic_power/coercion/mindslave
+	name =          "Mindslave"
+	cost =          28
+	cooldown =      200
+	use_melee =      TRUE
+	min_rank =      PSI_RANK_PARAMOUNT
+	use_description = "Grab a victim, target the eyes, then use the grab on them while on disarm intent, in order to convert them into a loyal mind-slave. The process takes some time, and failure is punished harshly."
+
+/datum/psionic_power/coercion/mindslave/invoke(var/mob/living/user, var/mob/living/target)
+	if(!istype(target) || user.zone_selected != BODY_ZONE_PRECISE_EYES || target.pulledby != user)
+		return FALSE
+	. = ..()
+	if(.)
+		if(target.stat == DEAD || (HAS_TRAIT(target, TRAIT_FAKEDEATH)))
+			to_chat(user, "<span class='warning'>\The [target] is dead!</span>")
+			return TRUE
+		if(!target.mind || !target.key)
+			to_chat(user, "<span class='warning'>\The [target] is mindless!</span>")
+			return TRUE
+		user.visible_message("<span class='danger'><i>\The [user] seizes the head of \the [target] in both hands...</i></span>")
+		to_chat(user, "<span class='warning'>You plunge your mentality into that of \the [target]...</span>")
+		to_chat(target, "<span class='danger'>Your mind is invaded by the presence of \the [user]! They are trying to make you a slave!</span>")
+		if(!do_after(user, target.stat == CONSCIOUS ? 80 : 40, target, 0, 1))
+			user.psi.backblast(rand(10,25))
+			return TRUE
+		to_chat(user, "<span class='danger'>You sear through \the [target]'s neurons, reshaping as you see fit and leaving them subservient to your will!</span>")
+		to_chat(target, "<span class='danger'>Your defenses have eroded away and \the [user] has made you their mindslave.</span>")
```

Comment:
.

Reply:
some do afters don't have a target after them, not sure that's what you want and could probably use the target in those do afters.... also use INTERACTING_WITH on checks to prevent do after spam